### PR TITLE
Remove rAF fork

### PR DESCRIPTION
**Pull request by flarnie**
**what is the change?:**
Undid https://github.com/facebook/react/pull/12837

**why make this change?:**
We originally forked rAF because we needed to pull in a particular
version of rAF internally at Facebook, to avoid grabbing the default
polyfilled version.

The longer term solution, until we can get rid of the global polyfill
behavior, is to initialize 'schedule' before the polyfilling happens.

Now that we have landed and synced
https://github.com/facebook/react/pull/12900 successfully, we can
initialize 'schedule' before the polyfill runs.
So we can remove the rAF fork. Here is how it will work:

1. Land this PR on Github.
2. Flarnie will quickly run a sync getting this change into www.
3. We delete the internal forked version of
   'requestAnimationFrameForReact'.
4. We require 'schedule' in the polyfill file itself, before the
   polyfilling happens.

**test plan:**
Flarnie will manually try the above steps locally and verify that things
work.

**issue:**
Internal task T29442940

### CODE REVIEW

**Comment by react-linter** [comment]
<details>
  <summary>Details of bundled changes.</summary>

  <p>Comparing: 36546b5137e9012ebdc62fc9ec11e3518c9e0aab...62567371dd0121a4d4d2c1617a8c79487ca8bc50</p>


  
## react-scheduler
File | Filesize Diff | Gzip Diff | Prev Size | Current Size | Prev Gzip | Current Gzip | ENV
 ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  ---  |  --- 
react-scheduler.development.js | -1.0% | -0.1% | 17.31 KB | 17.14 KB | 5.27 KB | 5.26 KB | UMD_DEV
react-scheduler.development.js | -1.0% | 0.0% | 17.12 KB | 16.95 KB | 5.22 KB | 5.22 KB | NODE_DEV
ReactScheduler-dev.js | +21.6% | +22.1% | 14.16 KB | 17.23 KB | 4.33 KB | 5.29 KB | FB_WWW_DEV
**ReactScheduler-prod.js** | **-0.4%** | **-0.6%** | **7.21 KB** | **7.19 KB** | **1.88 KB** | **1.87 KB** | **FB_WWW_PROD**

  </details>
  
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: packages/react-scheduler/src/ReactScheduler.js [hunk 1]
```diff
@@ -41,15 +41,30 @@ type CallbackConfigType = {|
...
+import warning from 'fbjs/lib/warning';
+
+if (__DEV__) {
+  if (
+    ExecutionEnvironment.canUseDOM &&
+    typeof requestAnimationFrame !== 'function'
+  ) {
+    warning(
+      false,
+      'schedule depends on requestAnimationFrame. Make sure that you load a ' +
```
Comment (gaearon) [inline packages/react-scheduler/src/ReactScheduler.js]: "schedule depends on requestAnimationFrame" doesn't make it obvious "schedule" is a name so it reads a bit odd. Maybe we can reword the whole message to not mention the library name? e.g.

>This browser doesn't support requestAnimationFrame. Make sure [...]

**Comment by gaearon** [comment]
(Will also need a Jest test fix)

**Comment by flarnie** [comment]
I've manually tested this and it seems ok - going to try and land and quickly sync early tomorrow morning.

**Comment by flarnie** [comment]
Found a potential issue internally, will debug that before landing this on Github. See internal task T29442940

**Comment by bvaughn** [comment]
I _think_ this changed causes a runtime error in the jsdom environment which doesn't occur in the latest release (16.4.1). I discovered this while testing the DevTools profiler on sites shared by open source users. One of those sites uses jsdom as part of the production build process.

**Comment by flarnie** [comment]
@bvaughn If you remember which site it was, message me a link?
This should only affect the Facebook build, and possibly test runs that are using async mode.

**Comment by gaearon** [comment]
Why does it only affect the Facebook build? Changes in the `scheduler` code would affect all builds.

Specifically, I think the issue might be that we warn when `typeof requestAnimationFrame !== 'function'`, but then *we read it anyway* (which would cause a reference error in strict mode if it's really not defined).

**Comment by gaearon** [comment]
https://github.com/facebook/react/pull/13152

**Comment by flarnie** [comment]
Thanks for the quick fix! I see that we always should guard when accessing a possibly not defined global API.

I know we wanted to avoid using `invariant` to avoid sending the bytes, reduce dependency, but I also wonder if it makes sense to throw an error if we reach that part of the code and `requestAnimationFrame` is not defined? Or maybe just at the point where we try to use it?

> Why does it only affect the Facebook build? Changes in the scheduler code would affect all builds.

The `scheduler` should only be used in async mode, and the only change there was a `DEV` only warning, hence my thinking that only tests running in async would fail, no production behavior would be expected to change. I see now that even requiring this module without a rAF polyfill could lead to an error, without the guarding you're adding.

**Pull request merged**